Wrong validation on accountSid when using API Sid/Api Key to initialize the client

# Issue Summary
When initialize the client using API Sid/Key, validation still applied to the username field which will check if the field starts with 'AC'. But API Sid should start with 'SK'.

### Steps to Reproduce
Initialize the twilio SDK client using API Sid/Key

### Code Snippet
```node
import twilio from 'twilio'
twilio(apiSid, apiKey)
```

### Exception/Log
```
Error: accountSid must start with AC
    at new Twilio (/var/task/node_modules/twilio/lib/rest/Twilio.js:137:11)
    at Object.initializer [as default] (/var/task/node_modules/twilio/lib/index.js:9:10)
    at new Twilio (/var/task/src/services/twilio.js:9:39)
    at initTwilio (/var/task/src/controllers/sms.js:19:16)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Object.sendVerification (/var/task/src/controllers/sms.js:49:28)
```

### Technical details:
* twilio-node version: 3.48.0
* node version: 13.12.0

`2021-01-21T12:09:44.457285+00:00 app[web.1]: 
2021-01-21T12:09:44.457303+00:00 app[web.1]: > hardcore-whatsapp@0.0.0 start /app
2021-01-21T12:09:44.457303+00:00 app[web.1]: > node ./bin/www
2021-01-21T12:09:44.457304+00:00 app[web.1]: 
2021-01-21T12:09:44.988266+00:00 app[web.1]: /app/node_modules/twilio/lib/rest/Twilio.js:140
2021-01-21T12:09:44.988280+00:00 app[web.1]:     throw new Error('accountSid must start with AC');
2021-01-21T12:09:44.988281+00:00 app[web.1]:     ^
2021-01-21T12:09:44.988282+00:00 app[web.1]: 
2021-01-21T12:09:44.988282+00:00 app[web.1]: Error: accountSid must start with AC
2021-01-21T12:09:44.988283+00:00 app[web.1]:     at new Twilio (/app/node_modules/twilio/lib/rest/Twilio.js:140:11)
2021-01-21T12:09:44.988284+00:00 app[web.1]:     at initializer (/app/node_modules/twilio/lib/index.js:9:10)
2021-01-21T12:09:44.988284+00:00 app[web.1]:     at Object.<anonymous> (/app/helpers/user-helper.js:5:33)
2021-01-21T12:09:44.988285+00:00 app[web.1]:     at Module._compile (internal/modules/cjs/loader.js:999:30)
2021-01-21T12:09:44.988285+00:00 app[web.1]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
2021-01-21T12:09:44.988285+00:00 app[web.1]:     at Module.load (internal/modules/cjs/loader.js:863:32)
2021-01-21T12:09:44.988286+00:00 app[web.1]:     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
2021-01-21T12:09:44.988286+00:00 app[web.1]:     at Module.require (internal/modules/cjs/loader.js:887:19)
2021-01-21T12:09:44.988286+00:00 app[web.1]:     at require (internal/modules/cjs/helpers.js:74:18)
2021-01-21T12:09:44.988287+00:00 app[web.1]:     at Object.<anonymous> (/app/routes/users.js:3:20)
2021-01-21T12:09:45.001711+00:00 app[web.1]: npm ERR! code ELIFECYCLE
2021-01-21T12:09:45.001965+00:00 app[web.1]: npm ERR! errno 1
2021-01-21T12:09:45.006786+00:00 app[web.1]: npm ERR! hardcore-whatsapp@0.0.0 start: `node ./bin/www`
2021-01-21T12:09:45.006883+00:00 app[web.1]: npm ERR! Exit status 1
2021-01-21T12:09:45.006997+00:00 app[web.1]: npm ERR! 
2021-01-21T12:09:45.007084+00:00 app[web.1]: npm ERR! Failed at the hardcore-whatsapp@0.0.0 start script.
2021-01-21T12:09:45.007173+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2021-01-21T12:09:45.015485+00:00 app[web.1]: 
2021-01-21T12:09:45.015632+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
2021-01-21T12:09:45.015720+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2021-01-21T12_09_45_007Z-debug.log
2021-01-21T12:09:45.089679+00:00 heroku[web.1]: State changed from starting to crashed
2021-01-21T12:09:45.056816+00:00 heroku[web.1]: Process exited with status 1
2021-01-21T12:31:31.720137+00:00 heroku[web.1]: State changed from crashed to starting
2021-01-21T12:31:43.265966+00:00 heroku[web.1]: Starting process with command `npm start`
2021-01-21T12:31:47.056171+00:00 app[web.1]: 
2021-01-21T12:31:47.056186+00:00 app[web.1]: > hardcore-whatsapp@0.0.0 start /app
2021-01-21T12:31:47.056186+00:00 app[web.1]: > node ./bin/www
2021-01-21T12:31:47.056187+00:00 app[web.1]: 
2021-01-21T12:31:48.289098+00:00 app[web.1]: /app/node_modules/twilio/lib/rest/Twilio.js:140
2021-01-21T12:31:48.289156+00:00 app[web.1]:     throw new Error('accountSid must start with AC');
2021-01-21T12:31:48.289157+00:00 app[web.1]:     ^
2021-01-21T12:31:48.289157+00:00 app[web.1]: 
2021-01-21T12:31:48.289158+00:00 app[web.1]: Error: accountSid must start with AC
2021-01-21T12:31:48.289158+00:00 app[web.1]:     at new Twilio (/app/node_modules/twilio/lib/rest/Twilio.js:140:11)
2021-01-21T12:31:48.289159+00:00 app[web.1]:     at initializer (/app/node_modules/twilio/lib/index.js:9:10)
2021-01-21T12:31:48.289159+00:00 app[web.1]:     at Object.<anonymous> (/app/helpers/user-helper.js:5:33)
2021-01-21T12:31:48.289159+00:00 app[web.1]:     at Module._compile (internal/modules/cjs/loader.js:999:30)
2021-01-21T12:31:48.289160+00:00 app[web.1]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
2021-01-21T12:31:48.289160+00:00 app[web.1]:     at Module.load (internal/modules/cjs/loader.js:863:32)
2021-01-21T12:31:48.289160+00:00 app[web.1]:     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
2021-01-21T12:31:48.289161+00:00 app[web.1]:     at Module.require (internal/modules/cjs/loader.js:887:19)
2021-01-21T12:31:48.289161+00:00 app[web.1]:     at require (internal/modules/cjs/helpers.js:74:18)
2021-01-21T12:31:48.289161+00:00 app[web.1]:     at Object.<anonymous> (/app/routes/users.js:3:20)
2021-01-21T12:31:48.353877+00:00 app[web.1]: npm ERR! code ELIFECYCLE
2021-01-21T12:31:48.366134+00:00 app[web.1]: npm ERR! errno 1
2021-01-21T12:31:48.413004+00:00 app[web.1]: npm ERR! hardcore-whatsapp@0.0.0 start: `node ./bin/www`
2021-01-21T12:31:48.413297+00:00 app[web.1]: npm ERR! Exit status 1
2021-01-21T12:31:48.413388+00:00 app[web.1]: npm ERR! 
2021-01-21T12:31:48.413578+00:00 app[web.1]: npm ERR! Failed at the hardcore-whatsapp@0.0.0 start script.
2021-01-21T12:31:48.413748+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2021-01-21T12:31:48.439932+00:00 app[web.1]: 
2021-01-21T12:31:48.440138+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
2021-01-21T12:31:48.440224+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2021-01-21T12_31_48_414Z-debug.log
2021-01-21T12:31:48.514843+00:00 heroku[web.1]: Process exited with status 1
2021-01-21T12:31:48.569054+00:00 heroku[web.1]: State changed from starting to crashed`


I am getting this error while deploying my application to heroku

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -117,7 +117,7 @@
     ],
     "quotes": [
       2,
-      "single"
+      "double"
     ],
     "indent": [
       2,

--- a/.jshintrc
+++ b/.jshintrc
@@ -44,7 +44,7 @@
   "newcap": true,
 
   // Enforce use of single quotation marks for strings.
-  "quotmark": "single",
+  "quotmark": "double",
 
   // Enforce placing 'use strict' at the top function scope
   "strict": true,

--- a/lib/base/BaseTwilio.ts
+++ b/lib/base/BaseTwilio.ts
@@ -88,7 +88,11 @@ export class BaseTwilio {
     }
 
     if (!this.accountSid?.startsWith("AC")) {
-      throw new Error("accountSid must start with AC");
+      const apiKeyMsg = this.accountSid.startsWith("SK")
+        ? ". The given SID indicates an API Key which requires the accountSid to be passed as an additional option"
+        : "";
+
+      throw new Error("accountSid must start with AC" + apiKeyMsg);
     }
   }
 

--- a/spec/unit/rest/Twilio.spec.js
+++ b/spec/unit/rest/Twilio.spec.js
@@ -1,17 +1,38 @@
 "use strict";
 const nock = require("nock");
-const util = require("util");
-var moduleInfo = require("../../../package.json");
-var os = require("os");
-var url = require("url"); /* jshint ignore:line */
 
 describe("client", () => {
-  var client;
-  const twilio = require("../../../lib");
+  let client;
+  const Twilio = require("../../../lib");
+
+  describe("initializing", () => {
+    it("should use the first arg for the username as well", () => {
+      client = new Twilio("ACXXXXXXXX", "test-password");
+      expect(client.username).toEqual("ACXXXXXXXX");
+      expect(client.accountSid).toEqual("ACXXXXXXXX");
+    });
+
+    it("should use the first arg for the username as well and the option as the accountSid", () => {
+      client = new Twilio("SKXXXXXXXX", "test-password", {
+        accountSid: "ACXXXXXXXX",
+      });
+      expect(client.username).toEqual("SKXXXXXXXX");
+      expect(client.accountSid).toEqual("ACXXXXXXXX");
+    });
+
+    it("should throw given an invalid accountSid", () => {
+      expect(() => new Twilio("ADXXXXXXXX", "test-password")).toThrow(
+        "must start with"
+      );
+      expect(() => new Twilio("SKXXXXXXXX", "test-password")).toThrow(
+        "API Key"
+      );
+    });
+  });
 
   describe("setting region and edge", () => {
     beforeEach(() => {
-      client = new twilio("ACXXXXXXXX", "test-password");
+      client = new Twilio("ACXXXXXXXX", "test-password");
     });
     describe("setting the region", () => {
       it("should use no region or edge by default", () => {
@@ -112,7 +133,7 @@ describe("client", () => {
 
   describe("adding user agent extensions", () => {
     it("sets the user-agent by default", () => {
-      const client = new twilio("ACXXXXXXXX", "test-password");
+      const client = new Twilio("ACXXXXXXXX", "test-password");
       const scope = nock("https://api.twilio.com", {
         reqheaders: {
           "User-Agent":
@@ -127,7 +148,7 @@ describe("client", () => {
     });
 
     it("allows for user-agent extensions", () => {
-      const client = new twilio("ACXXXXXXXX", "test-password", {
+      const client = new Twilio("ACXXXXXXXX", "test-password", {
         userAgentExtensions: [
           "twilio-run/2.0.0-test",
           "@twilio-labs/plugin-serverless/1.1.0-test",

